File: Practice/20/C++/20/20.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
	int money, n, k, b = 0;

	cin >> money >> n;

	vector<string> drinks(n);
	vector<int> price(n);
	vector<int> v(n);
	vector<int> rez(n);
	vector<int> liter(n);
	vector<int> arr(n);

	for (int i = 0; i < n; i++)
	{
		cin >> drinks[i] >> price[i] >> v[i];
		rez[i] = money / price[i];
		liter[i] = rez[i] * v[i];
		if (liter[i] == 0) b++;
	}

	if (b == n) {
		cout << -1 << endl;
		return(0);
	}

	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < n; j++)
		{
			if (liter[i] > liter[j]) k = i;
		}
	}

	cout << drinks[k] << " " << rez[k] << " " << endl << liter[k] << endl << money - (price[k] * rez[k]) << endl;

	return(0);
}

File: Practice/20/Python/20.py
b = 0
money, n = int(input()), int(input())
drinks = [0] * n
price = [0] * n
v = [0] * n
rez = [0] * n
liter = [0] * n
for i in range (n) :
    drinks[i], price[i], v[i] = map(str, input().split(' '))
    #drinks[i] = ''.join(drinks[i])
    #price[i] = ''.join(price[i])
    #v[i] = ''.join(v[i])
    #print(drinks[i], price[i], v[i])
    price[i] = int(price[i])
    v[i] = int(v[i])
    rez[i] = int(money / price[i])
    liter[i] = rez[i] * v[i]
    #print(rez[i], liter[i])
    if liter[i] == 0 : b = b + 1
if b == n :
    print(-1)
    exit(0)
for i in range (n) :
    for j in range (n) :
        if liter[i] > liter[j] : k = i
print(drinks[k], rez[k])
print(liter[k])
print(money - (price[k] * rez[k]))